building.rb

```diff
@@ -1,2 +1,28 @@
 class Building
+  attr_reader :apartments
+  
+  def initialize(args = {})
+    @apartments = args.fetch(:apartments, [])
+    @address = args.fetch(:address, "N/A")
+  end
+
+  def total_room_count
+    @apartments.reduce(0) {|sum, apartment| sum + apartment.room_count}
+  end
+
+  def total_monthly_revenue
+    @apartments.inject(0) {|sum, apartment| sum + apartment.monthly_rent}
+  end
+
+  def apartments_by_rent
+    @apartments.sort_by {|apartment| apartment.monthly_rent}.reverse
```

Comment:
Since monthly rent is a number, you can just make it negative to get your apartments sorted in the correct order without having to sort them and then call reverse on the sorted array.
```ruby
@apartments.sort_by { |apartment| -apartment.monthly_rent }
```